projects/archiver/zipper.ts

```diff
@@ -0,0 +1,66 @@
+import { S3, SharedIniFileCredentials } from 'aws-sdk'
+import { PassThrough } from 'stream'
+import archiver = require('archiver')
+
+const creds = process.env.arn
+    ? {}
+    : {
+          credentials: new SharedIniFileCredentials({ profile: 'frontend' }),
+      }
+const s3 = new S3({
+    region: 'eu-west-1',
+    ...creds,
+})
+
+const notNull = <T>(value: T | null | undefined): value is T =>
+    value !== null && value !== undefined
+
+const removeInitial = (s: string, remove: string) => {
+    if (!s.startsWith(remove)) return s
+    return s.substr(remove.length)
+}
+
+export const zip = async (name: string, root: string, prefix: string) => {
```

Comment:
Could we type `root` as something a bit more refined just to avoid typos, e.g. `'data' | ...`